src/gcode_util.js

```diff
@@ -102,7 +102,7 @@ function extractCommandSequence(text) {
     }
 
     // Check for the start of a new command.
-    if (c == 'G' || c == 'M') {
+    if (c == 'G' || c == 'M' || c == '$') {
```

Comment:
While it doesn't necessarily make sense to want to do it during a job, grbl does permit a '$$' command to display configuration information, but splitting like this would prevent that command from being sent properly.

I've been thinking about your use case of overwriting machine configurations during a job; I think the code as is would still permit you to send the $ commands, they would just be grouped with the prior command, left to your controller board to parse apart.

Reply:
The problem that I'm seeing is that the parser removes the newlines, which GRBL doesn't like - the sent command looks like `G90 $120=25 $121=25`, to which GRBL sends `error:1`. Splitting the $ codes into their own lines fixes this.

I have a workaround for this (install my patched extension locally), so if it's difficult to deploy this and would be more complex to make it work, it's probably not worth pursuing.